spec_summary/filter_map.go

```diff
@@ -37,18 +37,27 @@ func NewFiltersToMethods[MethodID comparable]() *FiltersToMethods[MethodID] {
 	}
 }
 
-// Registers the given method as having the given value for the given filter
-// kind.
-func (fm *FiltersToMethods[MethodID]) InsertNondirectionalFilter(filter FilterKind, value string, method MethodID) {
+// Registers the given filter and value in the map. These are also associated
+// with the method ID if it exists in the given set of allowed methods.
+func (fm *FiltersToMethods[MethodID]) InsertNondirectionalFilter(filter FilterKind, value string, method MethodID, allowedMethods sets.Set[MethodID]) {
```

Comment:
This is fine, but I think it would be a little cleaner to change the type of `method` to `optionals.Optional[MethodID]` and let the caller decide when it should be Some or None.

Reply:
This PR mirrors the changes in #168. My initial approach there was indeed to pass option values, but I found myself duplicating the same code at most of the callers, so I figured I'd push it down into the implementation here.

Reply:
I've opened akitasoftware/go-utils#11 to factor that common code into a method on `sets.Set`.

Reply:
In 534259a.